Source/Android/jni/SkylanderConfig.cpp

```diff
@@ -0,0 +1,179 @@
+
+    jobject pairObj = env->NewObject(pairClass, pairInit, intId, varId);
+
+    env->CallObjectMethod(hashMapObj, hashMapOut, pairObj, ToJString(env, name));
+  }
+
+  return hashMapObj;
+}
+
+extern "C" JNIEXPORT jobject JNICALL
+Java_org_dolphinemu_dolphinemu_features_skylanders_SkylanderConfig_getInverseSkylanderMap(
+    JNIEnv* env, jclass clazz)
+{
+  jclass hashMapClass = env->FindClass("java/util/HashMap");
+  jmethodID hashMapInit = env->GetMethodID(hashMapClass, "<init>", "(I)V");
+  jobject hashMapObj = env->NewObject(hashMapClass, hashMapInit,
+                                      static_cast<u16>(IOS::HLE::USB::list_skylanders.size()));
+  jmethodID hashMapOut = env->GetMethodID(
+      hashMapClass, "put", "(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;");
+
+  jclass pairClass = env->FindClass("android/util/Pair");
+  jmethodID pairInit =
+      env->GetMethodID(pairClass, "<init>", "(Ljava/lang/Object;Ljava/lang/Object;)V");
+
+  jclass integerClass = env->FindClass("java/lang/Integer");
+  jmethodID intInit = env->GetMethodID(integerClass, "<init>", "(I)V");
+
+  for (auto it : IOS::HLE::USB::list_skylanders)
+  {
+    std::string name = it.second;
+    u16 id = it.first.first;
+    u16 var = it.first.second;
+
+    jobject intId = env->NewObject(integerClass, intInit, static_cast<jint>(id));
+    jobject varId = env->NewObject(integerClass, intInit, static_cast<jint>(var));
+
+    jobject pairObj = env->NewObject(pairClass, pairInit, intId, varId);
+
+    env->CallObjectMethod(hashMapObj, hashMapOut, ToJString(env, name), pairObj);
+  }
+
+  return hashMapObj;
+}
+extern "C" JNIEXPORT jboolean JNICALL
+Java_org_dolphinemu_dolphinemu_features_skylanders_SkylanderConfig_removeSkylander(JNIEnv* env,
+                                                                                   jclass clazz,
+                                                                                   jint slot)
+{
+  auto& system = Core::System::GetInstance();
+  return static_cast<jboolean>(system.GetSkylanderPortal().RemoveSkylander(slot));
+}
+extern "C" JNIEXPORT jobject JNICALL
+Java_org_dolphinemu_dolphinemu_features_skylanders_SkylanderConfig_loadSkylander(JNIEnv* env,
+                                                                                 jclass clazz,
+                                                                                 jint slot,
+                                                                                 jstring file_name)
+{
+  File::IOFile sky_file(GetJString(env, file_name), "r+b");
+  if (!sky_file)
+  {
+    return nullptr;
+  }
+  std::array<u8, 0x40 * 0x10> file_data;
+  if (!sky_file.ReadBytes(file_data.data(), file_data.size()))
+  {
+    return nullptr;
+  }
+
+  u16 sky_id = file_data[0x11];
+  u16 sky_var = file_data[0x1D];
+  sky_id <<= 8;
+  sky_var <<= 8;
+  sky_id |= file_data[0x10];
+  sky_var |= file_data[0x1C];
```

Comment:
Consider putting this bit math outside of the frontend, so it doesn't need to be duplicated between DolphinQt and Android.